sdks/how-to/how-to-share-data.md

```diff
@@ -0,0 +1,162 @@
+---
+slug: how-to-share-data
+---
+
+# Sharing data between data owners
+
+In iCure it is possible to allow multiple data owners to access specific instances of `Patient`, `DataSample` and 
+`HealthcareElement`.
+
+Initially, when a data owner creates one of these entities only whey will be able to access it. 
+However, it is possible for the creator to share the entity with others by using the `giveAccessTo` methods in the 
+respective apis.
+This way also the other data owners will be able to retrieve the entity and decrypt its content.
+
+## Instantiate the apis for different users
+
+In the following examples we will use three different instances of `MedTechApi`s, to perform the requests as different
+users.
+The api we use are `hcp1Api` and `hcp2Api` to act as two different healthcare practitioners data owners (`hcp1` and 
+`hcp2`, respectively) and `pApi` to act as a patient data owner (`p`).
+
+## Sharing data other users
+
+First create a new entity, in this example `hcp1` creates a patient:
+
+<!-- file://code-samples/how-to/sharing-data/index.mts snippet:create a patient-->
+```typescript
+// hcp1 creates a new patient
+const note = 'Winter is coming'
+const patient = await hcp1Api.patientApi.createOrModifyPatient(
+  new Patient({ firstName: 'John', lastName: 'Snow', note }),
+)
+```
+
+`hcp1` can now share the newly created patient with other data owners, no matter if they are other healthcare 
+practitioners, patients or devices.
+
+To do this `hcp1` simply has to use `giveAccessTo` method from the `patientApi`.
+This method takes two parameters: the entity that will be shared (in the example `patient`), and the id of the data 
+owner that the entity will be shared with.
+
+<!-- file://code-samples/how-to/sharing-data/index.mts snippet:share a patient-->
+```typescript
+// hcp1 shares the information of `patient` with hcp2
+const updatedPatient = await hcp1Api.patientApi.giveAccessTo(
+  patient,
+  hcp1Api.dataOwnerApi.getDataOwnerIdOf(hcp2),
+)
+// hcp1 shares the information of `patient` with p (a different patient that is also a data owner)
+await hcp1Api.patientApi.giveAccessTo(
+  updatedPatient,
+  hcp1Api.dataOwnerApi.getDataOwnerIdOf(p),
+)
+```
+
+:::caution
+
+The `giveAccessTo` method **modifies** the shared entity and will fail if the version of the input entity doesn't 
+match the latest version.
+
+In the previous example we shared the new `patient` first with `hcp2`, then with `p`. 
+Since the patient changes we can't pass again `patient` to the second invocation of `giveAccessTo`, but we have to pass 
```

Comment:
```suggestion
Since the first call modifies the patient entity, we cannot pass again `patient` to the second invocation of `giveAccessTo`, but we have to pass 
```